geomstats/geometry/euclidean.py

```diff
@@ -85,10 +86,9 @@ def metric_matrix(self, base_point=None):
         inner_prod_mat : array-like, shape=[..., dim, dim]
             Inner-product matrix.
         """
-        mat = gs.eye(self._space.dim)
-        if base_point is not None and base_point.ndim > 1:
-            mat = gs.broadcast_to(mat, base_point.shape + (self._space.dim,))
-        return mat
+        dim = self._space.dim
+        mat = gs.eye(dim)
+        return repeat_out(self._space, mat, base_point, out_shape=(dim, dim))
```

Comment:
NIT: mat could be called `out` for consistency with how the function repeat_out is used elsewhere.

Alternatively (probably preferred), avoid using "out" as a variable and call each variable by its mathematical interpretation?